src/components/forms/Datepicker.tsx

```diff
@@ -16,13 +16,20 @@ import 'react-datepicker/dist/react-datepicker.css';
 type Props = CommonProps;
 
 export function Datepicker(props: Props) {
-  const [date, setDate] = useState<Date>(new Date());
-
+  const [date, setDate] = useState<Date>(props.value?new Date(props.value):new Date());
+console.log(props.value)
   return (
     <ReactDatePicker
       selected={date}
       className={`w-full py-2 px-3 rounded border border-gray-300 text-sm ${props.className}`}
-      onChange={(date) => setDate(date as Date)}
+      onChange={(date) => {
+        setDate(date as Date)
+        props.onChange(date?.toISOString()
+        .split('T')[0])
```

Comment:
This is somehow hard to understand, just looking at the code. Can we restructure this with proper types?